core/src/main/scala/org/apache/spark/sql/delta/files/DeltaSourceSnapshot.scala

```diff
@@ -52,51 +54,45 @@ class DeltaSourceSnapshot(
     (part, data)
   }
 
-  protected def initialFiles: Dataset[IndexedFile] = {
+  lazy val initialFiles: DataFrame = {
```

Comment:
The correct use of `cacheDS()` is (see e.g. `Snapshot.cachedState`):
```scala
private lazy val cachedState = {
    ...
  cacheDS(...)
}

protected def initialFiles: Dataset[IndexedFile] = cacheDS.getDF
```
(`StateCache` handles the various thread races that might arise, so we can just let its do its job)

Reply:
I can update to that. The current method was wrong because it requires rerunning zipWithIndex (which immediately triggers jobs) before realizing the result was cached

Reply:
Actually, looking closer -- why are we caching `initialFiles` in the first place, when the dataframe we actually care about is returned by `DeltaLog.filterFileList` below? Should we cache _that_ instead?

Reply:
Kinda related to my other comment that I don't think partitionFilters is ever actually defined or passed in anywhere?

Reply:
I just went ahead and removed everything filter related since it was never actually used anywhere

Reply:
Fair, but if anyone ever fixed that shortcoming this code would be a hidden performance hazard, no?
(cached datasets don't benefit from predicate pushdown)

Reply:
I could add it back in but it's just dead code. I can't think how filters would even ever get to the DeltaSourceSnapshot as you don't have any context of the query involved in the stream AFAIK. I can only imagine this maybe was used when Delta "supported" (but warned you was deprecated) reading a path that includes a partition. So if you tried to stream from the delta table at `/path/to/table` using the path `/path/to/table/year=2023`, maybe that got pushed to the `DeltaSource` but that functionality was removed?